datafusion/functions/src/datetime/date_part.rs

```diff
@@ -127,7 +127,12 @@ impl ScalarUDFImpl for DatePartFunc {
             ColumnarValue::Scalar(scalar) => scalar.to_array()?,
         };
 
-        let arr = match part.to_lowercase().as_str() {
+        let arr = match part
+            .to_lowercase()
+            .replacen('\'', "", 2)
+            .replacen('\"', "", 2)
+            .as_str()
```

Comment:
Hmm, this will still let `select extract("'epoch'" from now());` through

But this is quite a minor thing, not sure how much it matters in the wider picture :sweat_smile:

Reply:
I found a workaround, but it should be a better way to handle it in sqlparser. Because it should have many use cases.